src/Chainweb/BlockHeaderDB/RestAPI/Server.hs

```diff
@@ -276,3 +295,35 @@ someBlockHeaderDbServer (SomeBlockHeaderDb (db :: BlockHeaderDb_ v c))
 someBlockHeaderDbServers :: ChainwebVersion -> [(ChainId, BlockHeaderDb)] -> SomeServer
 someBlockHeaderDbServers v = mconcat
     . fmap (someBlockHeaderDbServer . uncurry (someBlockHeaderDbVal v))
+
+-- -------------------------------------------------------------------------- --
+-- BlockHeader Event Stream
+
+someHeaderStreamServer :: PayloadCas cas => ChainwebVersion -> CutDb cas -> SomeServer
+someHeaderStreamServer (FromSing (SChainwebVersion :: Sing v)) cdb =
+    SomeServer (Proxy @(HeaderStreamApi v)) $ headerStreamServer cdb
+
+headerStreamServer
+    :: forall cas (v :: ChainwebVersionT)
+    .  PayloadCas cas
+    => CutDb cas
+    -> Server (HeaderStreamApi v)
+headerStreamServer cdb = headerStreamHandler cdb
+
+headerStreamHandler :: forall cas. PayloadCas cas => CutDb cas -> Tagged Handler Application
+headerStreamHandler db = Tagged $ \req respond -> do
+    chan <- newChan
+    void . SP.mapM_ (g >=> writeChan chan . f) . SP.concat $ blockDiffStream db
```

Comment:
- What does `Tagged` do here? 

- `void` is unnecessary 

- Why split `g` and `f`? Kleisli fish are nice and all but you've defined the functions locally below, and there doesn't seem to be any multiple usages.

Reply:
The endpoint that this handler corresponds to ends in `Raw`. This is the strategy that we've discovered for doing Server-Sent Events with Servant.

`void` was indeed necessary.

`f` and `g` are split for cleanliness.